### DISCUSSION TITLE
Only show custom focus representation

### DISCUSSION THREAD

**Discussion by rdirisio** [molstar/molstar#902] [Q&A]
Hello, 

I wrote a `CustomStructureFocusRepresentation` that I want to use when focusing on a custom system that I constructed with `StructureRepresentationPresetProvider`.  It only includes surroundings and non-covalent interactions. 

Currently, I only use `CustomStructureFocusRepresentation` in my initial Behaviors when I create the plugin:
```
const MySpec: PluginUISpec = {
  ...DefaultPluginUISpec(),
  layout: {
    initial: {
      isExpanded: false,
      showControls: true,
      controlsDisplay: "reactive" as PluginLayoutControlsDisplay,
    },
  },
  behaviors: [
    PluginSpec.Behavior(CustomStructureFocusRepresentation), //replaced StructureFocusRepresentation here
    PluginSpec.Behavior(PluginBehaviors.Representation.HighlightLoci),
    PluginSpec.Behavior(PluginBehaviors.Representation.SelectLoci),
    PluginSpec.Behavior(PluginBehaviors.Representation.DefaultLociLabelProvider),
    PluginSpec.Behavior(PluginBehaviors.Representation.FocusLoci),
    PluginSpec.Behavior(PluginBehaviors.Camera.FocusLoci),
    PluginSpec.Behavior(PluginBehaviors.Camera.CameraAxisHelper),
    PluginSpec.Behavior(PluginBehaviors.Camera.CameraControls),
    PluginSpec.Behavior(PluginBehaviors.CustomProps.StructureInfo),
    PluginSpec.Behavior(PluginBehaviors.CustomProps.AccessibleSurfaceArea),
    PluginSpec.Behavior(PluginBehaviors.CustomProps.BestDatabaseSequenceMapping),    
    PluginSpec.Behavior(PluginBehaviors.CustomProps.Interactions),
    PluginSpec.Behavior(PluginBehaviors.CustomProps.SecondaryStructure),
    PluginSpec.Behavior(PluginBehaviors.CustomProps.ValenceModel),
    PluginSpec.Behavior(PluginBehaviors.CustomProps.CrossLinkRestraint),
  ],
};
```
But, when I click on my custom system, I see the original `StructureFocusRepresentation` focus components in my state tree (Target and Surroundings) as well as my custom one. My question is: how do I only show my custom one? If I comment out the line:
```
    PluginSpec.Behavior(CustomStructureFocusRepresentation), //replaced StructureFocusRepresentation here
``` 
I still see the original focus representation. It seems that `PluginBehaviors.Representation.FocusLoci` is toggling this focus on and off.

Do I need to override the StructureFocusRepresentation in my `StructureRepresentationPresetProvider`? Or somewhere else? If so, how do I go about doing so? 

I see that the representation-presets use `updateFocusRepr`, but that defaults to `StructureFocusRepresentation`.

Any advice is appreciated. Thanks!

**Comment by dsehnal** [comment]
This is an unintended side effect of updating the focus representation from [inside the representation presets](https://github.com/molstar/molstar/blob/9ac204cb6e7d501deac3e02ca058301873f03884/src/mol-plugin-state/builder/structure/representation-preset.ts#L203).

I will see about making a PR which doesn't call this code if the behavior wasn't added.

**Reply by rdirisio** [reply]
Fantastic, thank you. I see https://github.com/molstar/molstar/pull/903 was merged. I will use the new feature when a new version is pushed 👍
